src/Data/Array/Polarized/Pull.hs

```diff
@@ -1,21 +1,85 @@
 {-# LANGUAGE LinearTypes #-}
 {-# LANGUAGE NoImplicitPrelude #-}
 
--- | This module is intended to be imported qualified, e.g.
+-- | This module provides pull arrays.
+--
+-- Use pull arrays to force fusion (i.e., force GHC to not allocate arrays in
+-- the garbage collector memory) in a computation that only uses lists
+-- as /intermediate/ results without operations that \"write\" to a list.
+-- You should think about pull arrays as arrays you can \"pull\" or read from,
+-- zip with and map over. These are arrays that work nicely as arguments to
+-- functions and poorly as arrays to write to.
```

Comment:
It's not that they work “poorly” as returned values, they are just less flexible there. If you have a pull array argument, you have quite a bit of the power of an allocated array. Whereas if you need to return one, you have quite a bit of restrictions.